input.go

```diff
@@ -94,17 +94,48 @@ func (i *Input) UnmarshalJSON(b []byte) error {
 		return err
 	}
 	i.UnlockingScript = s
-	i.PreviousTxIDBytes = ptxID
+	i.previousTxID = ptxID
 	i.PreviousTxOutIndex = ij.Vout
 	i.SequenceNumber = ij.Sequence
 	return nil
 }
 
+// PreviousTxIDAdd will add the supplied txID bytes to the Input,
+// if it isn't a valid transaction id an ErrInvalidTxID error will be returned.
+func (i *Input) PreviousTxIDAdd(txID []byte) error {
```

Comment:
just a thought: would it be better to force a 32 byte variable as input to the func and not return an error? so something like: `PreviousTxIDAdd(txID [32]byte)`?

Reply:
I could do, but fixed length arrays can be a wee bit annoying to work with, probably better that we just return an error because handling the error is easier to deal with